doc/decisions/0001-record-architecture-decisions.md

```diff
@@ -0,0 +1,45 @@
+# 1. Record architecture decisions
+
+Date: 2018-12-05
+
+## Status
+
+Accepted
+
+## Context
+
+We need to record the architectural decisions made on this project. A lot of the reasoning behind decisions made historically on this project have been lost, and we risk perpetuating that situation unless we record the conversations we are having around decisions we make.
+
+## Decision
+
+We will use Lightweight Architecture Decision Records, as described by [Michael Nygard](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions) and [ThoughtWorks](https://www.thoughtworks.com/radar/techniques/lightweight-architecture-decision-records).
+
+A template can be found at `doc/decisions/0000-template.md`, which is slightly modified version of the file generated by Nat Pryce's [Lightweight ADR toolset](https://github.com/npryce/adr-tools).
+
+We should only record "significant" decisions, not every single thing we do. Some examples of "significant" decisions might be:
+
+1) Adding new infrastructure to the service.
+3) Fundamental changes to the data model.
+2) Removing a feature or capability.
+
+"Significant" decisions could include those where we opt *not* to undertake a piece of work (e.g. adding infrastructure) in lieu of a more simple solution.
+
+All developers will be responsible for writing architectural decision records when they recognise it is appropraite to create a durable record of the context and conversation around a particular piece of work. All other developers will be responsible for reading, commenting on, and updating architectural decision records as appropriate. Conversations should take place on the GitHub Pull Request rather than other mediums, as this will keep all of the relevant points discoverable. Key points raised in discussion should be integrated into the architectural decision record, if appropriate.
```

Comment:
This is a *really* long line. Some of the others are pretty long too.

Maybe ADR Number 2 can be a proposal from me to "include line breaks at regular intervals in markdown files" :trollface: 

Could we wrap at 120 chars like our Python file rule?

Reply:
Ha, I actually spent more time than I'd like to admit breaking and then re-joining the lines and working out which I preferred. Manual line breaks is probably more consistent with our python, but it felt kinda painful to break long markdown lines, and then if you do, you end up with weirdly short lines if you have links on their own line, which is probably more broken?